prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -1710,6 +1710,14 @@ subroutine initProgramVariables(input, env)
     end if
     call TElectronicSolver_init(electronicSolver, input%ctrl%solver%iSolver, nBufferedCholesky)
 
+    if (any(electronicSolver%iSolver ==&
+        & [electronicSolverTypes%elpa, electronicSolverTypes%omm, electronicSolverTypes%pexsi,&
+        & electronicSolverTypes%ntpoly]) .and.&
+        & input%ctrl%parallelOpts%nGroup /= nIndepHam * nKPoint) then
+
```

Comment:
Isn't the first part of the condition equivalent to `electronicSolver%isElsiSolver`? That would be more compact and future safe.

Reply:
Not quite, as the ELPA solver doesn't fall into this category (but shouldn't get this far as its not used in sparse mode).